app/src/main/java/fr/neamar/kiss/db/DBHelper.java

```diff
@@ -137,6 +137,32 @@ private static Cursor getHistoryByAdaptive(SQLiteDatabase db, int hours, int lim
         return db.rawQuery(sql, null);
     }
 
+    /**
+     * Get the history items used closest to this time of day, for each day old an item is it has
+     * one less hour of time weight. So we limit the number of days of history to 24 days in the
+     * WHERE clause, this should also help with speed on large histories.
+     *
+     * This is done by taking the max of a triangle waveform whose period is 24 hours, amplitude
+     * is half the milliseconds in a day and begins at currentTimeMillis() - timestamp, then offset
+     * by the time difference / 48 to diminish older history items by an hour for every day old. 48
+     * is used because the triangle wave is half amplitude (1 / 2) * (1 / 24) = 1 / 48.
+     *
+     * @param db    The SQL db
+     * @param limit Maximum result size
+     * @return Cursor
+     */
+    private static Cursor getHistoryByTime(SQLiteDatabase db, int limit) {
+        long now = System.currentTimeMillis();
+        String sql = "SELECT record, MAX(ABS((" + now + " - timestamp) % 86400000 - 43200000) - (" + now + " - timestamp) / 48 ) AS value" +
+                " FROM history" +
+                " WHERE timestamp > " + now + " - 2073600000" +
```

Comment:
This may be written as `" WHERE timestamp > " + (now - 2073600000) + ` so there is less to calculate in query itself.

Also wondering where 2073600000 is coming from?
It seems pretty close to `Integer.MAX_VALUE` and value column may also contain pretty high numbers.
But with given values there will never be overflow when setting `ValuedHistoryRecord.value` 🤔

Reply:
You are right about that. Is sqlite slower? Since its outside the aggregate function I think it will still be only calculated once either way. I made sqlite do the heavy lifting to avoid long java types, and avoid overflow. The timestamps themselves are very large. 

The value is 24 days in millis. 

As for ValuedHistoryRecord.value I didnt see it actually used for anything. It seems the history items are sorted by relevance which is assigned by the order of the returned items correct? Either way maybe changing its type to long might be in order.

Reply:
I think it should be a constant keep in Java so we can better understand why it's used.

Reply:
I was able to prove out that ValuedHistoryRecord.value is not used in the getHistory() context, unless I'm missing something. HistroySearcher calls DataHandler.getHistory() which then calls DBHelper.getHistory(). DataHandler.getHistory() returns a List of Pojos which never reads the value from the List of ValuedHistoryRecord returned by DBHelper.getHistory(). I also added the value in the debug pop-up menu and Pojo.value is always 0.  When searching for usages of Pojo.value it appears to never be used. 

I found one usage of ValuedHistoryRecord.value in QuerySearch.doInBackground(). DBHelper.getPreviousResultsForQuery() returns a list of ValuedHistoryRecord that is added to the knownIds HashMap. Then when addResults is called relevance is modified by adding 25 * value. In this context ValuedHistoryRecord.value is simply an aggregate count(*). 

With all that said I think it's safe if ValuedHistoryRecord.value overflows for getHistory() since it appears to be used only in a search context.

Reply:
Should we close this review and open another one since its now outdated?